### ISSUE TITLE
[LibreOffice] build fails on x86_gcc2

### ISSUE THREAD

**Issue by diversys** [haikuports/haikuports#2810]
`undefined reference to __stack_chk_fail_local` is normally fixed by adding  `-fno-stack-protector`
which was added in bb0cc2b8eb1525b612507fca88db472248dd5748. However LO build still fails later on with this error:

```
[DEP] LNK:Library/libgcc3_uno.so
[LNK] Library/libgcc3_uno.so
[DEP] LNK:Library/libuno_purpenvhelpergcc3.so.3
[LNK] Library/libuno_purpenvhelpergcc3.so.3
/sources/core/workdir/CxxObject/bridges/source/cpp_uno/gcc3_linux_intel/callvirtualmethod.o: In function `.L13':
callvirtualmethod.cxx:(.text+0xe8): undefined reference to `__stack_chk_fail_local'
/sources/core/workdir/CxxObject/bridges/source/cpp_uno/shared/cppinterfaceproxy.o: In function `(anonymous namespace)::getVtableFactory()':
cppinterfaceproxy.cxx:(.text+0x110): undefined reference to `__stack_chk_fail_local'
/sources/core/workdir/CxxObject/bridges/source/cpp_uno/shared/cppinterfaceproxy.o: In function `bridges::cpp_uno::shared::CppInterfaceProxy::create(bridges::cpp_uno::shared::Bridge*, _uno_Interface*, _typelib_InterfaceTypeDescription*, rtl::OUString const&)':
cppinterfaceproxy.cxx:(.text+0x373): undefined reference to `__stack_chk_fail_local'
/sources/core/workdir/CxxObject/bridges/source/cpp_uno/shared/cppinterfaceproxy.o: In function `bridges::cpp_uno::shared::CppInterfaceProxy::acquireProxy()':
cppinterfaceproxy.cxx:(.text+0x42e): undefined reference to `__stack_chk_fail_local'
/sources/core/workdir/CxxObject/bridges/source/cpp_uno/shared/cppinterfaceproxy.o: In function `(anonymous namespace)::rtl_Instance<bridges::cpp_uno::shared::VtableFactory, (anonymous namespace)::InitVtableFactory, osl::Guard<osl::Mutex>, osl::GetGlobalMutex, int, int>::create((anonymous namespace)::InitVtableFactory, osl::GetGlobalMutex)':
cppinterfaceproxy.cxx:(.text+0x6d6): undefined reference to `__stack_chk_fail_local'
collect2: error: ld returned 1 exit status
/sources/core/bridges/Library_cpp_uno.mk:10: recipe for target '/sources/core/instdir/program/libgcc3_uno.so' failed
make[1]: *** [/sources/core/instdir/program/libgcc3_uno.so] Error 1
make[1]: *** Waiting for unfinished jobs....
Makefile:286: recipe for target 'build' failed
make: *** [build] Error 2
```
Most likely this flag is not used at this point for some reason.

http://vmpkg.haiku-os.org/master/x86_gcc2/logviewer.html?buildruns/1531/builds/28590.log

**Comment by KapiX** [comment]
`-fno-stack-protector` is not a solution. I had similar problems with 64-bit builds, and I fixed them with `-lssp`:
https://cgit.freedesktop.org/libreoffice/core/tree/solenv/gbuild/platform/haiku.mk#n14

Try building with `make verbose=t` to see the command line, and check if `-lssp` is there.

**Comment by alaviss** [comment]
I don't believe adding `-lssp` is the correct solution. The correct one should be to add `-fstack-protector` to link flags, which will make `libssp` or it's equivalent linked, and will work for both clang and gcc.

If that doesn't work, then there's something wrong with our gcc and should be fixed.

**Comment by KapiX** [comment]
If `-fstack-protector` wasn't there we wouldn't have this problem in the first place.
https://cgit.freedesktop.org/libreoffice/core/tree/configure.ac#n5824

**Comment by KapiX** [comment]
Fixed in 87139540d7c1c1762c23361d1bee92d0bbf3eec2.
